addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/ApplianceGroheOndusHandler.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.groheondus.internal.handler;
+
+import static org.openhab.binding.groheondus.internal.GroheOndusBindingConstants.*;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.grohe.ondus.api.OndusService;
+import org.grohe.ondus.api.model.Appliance;
+import org.grohe.ondus.api.model.ApplianceCommand;
+import org.grohe.ondus.api.model.ApplianceData;
+import org.grohe.ondus.api.model.ApplianceData.Measurement;
+import org.grohe.ondus.api.model.Location;
+import org.grohe.ondus.api.model.Room;
+import org.openhab.binding.groheondus.internal.ApplianceGroheOndusConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florian Schmidt and Arne Wohlert - Initial contribution
+ */
+@NonNullByDefault
+public class ApplianceGroheOndusHandler extends BaseThingHandler {
```

Comment:
Imho the better name would be `GroheOndusApplianceHandler`.

Reply:
Done